backend/src/services/user-service.ts

```diff
@@ -259,6 +312,33 @@ export class UserService implements IUserService {
     }
   }
 
+  /**
+   * Checks the user's password reset.
+   * @param email The user's email
+   * @param password The user's password
+   */
+  public async verifyUserResetPassword(
+    password: string,
+    token: string,
+  ): Promise<User | undefined> {
+    const user = await this._users.findOne({
+      where: {
+        forgotPasswordToken: token,
+      },
+    });
+
+    if (!user) {
+      throw new BadRequestError("invalid token");
+    }
+
+    if (user.forgotPasswordToken === token) {
```

Comment:
Das if könnte man sich sparen, oder? In der query wird ja schon nach dem token gesucht

Reply:
ja ich weiß, hab ich auch schon überlegt, irwie meine Paranoia bei dem psw reset.